crates/torii/core/src/engine.rs

```diff
@@ -574,16 +581,14 @@ impl<P: Provider + Sync> Engine<P> {
         transaction_hash: Felt,
     ) -> Result<()> {
         self.db.store_event(event_id, event, transaction_hash, block_timestamp);
-        let mut unprocessed = true;
-        for processor in &self.processors.event {
-            // If the processor has no event_key, means it's a catch-all processor.
-            // We also validate the event
-            if (processor.event_key().is_empty()
-                || get_selector_from_name(&processor.event_key())? == event.keys[0])
-                && processor.validate(event)
-            {
-                unprocessed = false;
-                if let Err(e) = processor
+        let event_key = event.keys[0];
+
+        let Some(processor) = self.processors.event.get(&event_key) else {
```

Comment:
We should also validate non catch all event processors, just like we did before. And we should also try to keep the behaviour the same between the catch all & norm al processors. We should return the catch all processor and then process it further down

Reply:
the reason i removed validation from them is since they are emitted from the world we can be sure that they conform with the structure that we expect, we would be unnecessarily doing validation for events already specified in the world. is there any case where validation is required for world non-catch all events?